apps/dashboard/app/views/widgets/_xdmod_widget_job_efficiency.html.erb

```diff
@@ -19,9 +19,9 @@
       {{else}}
       <p class="d-flex justify-content-between card-text font-weight-bold"><span class="text-success">{{good_percent}}% efficient</span> <span class="text-danger">{{bad_percent}}% inefficient</span></p>
       <div class="progress progress-custom">
-        <div class="progress-bar bg-success" role="progressbar" aria-label="percent efficient" aria-valuenow="{{good_percent}}" aria-valuemin="0" aria-valuemax="100" style="width: {{good_percent}}%">
+        <div class="progress-bar bg-success w-{{good_percent}}" role="progressbar" aria-label="percent efficient" aria-valuenow="{{good_percent}}" aria-valuemin="0" aria-valuemax="100">
```

Comment:
Unsure if this is valid. Don't know where this value comes from. Functions in lines 79 and 82 have been changed to round to whole number, but I think these values are templated in from elsewhere. If this is a javascript-enabled template we could use `Math.round(good_percent)`, `good_percent.round` if ruby.

Reply:
Right, that's already been done. How do you turn on CSP warnings?

Reply:
>  I'm now trying to investigate other CSP errors. How do you turn on CSP warnings? 

You'll need a combination of these configs (use your own directory, this is mine)
```
# .env.local
OOD_CONFIG_D_DIRECTORY="/users/PZS0714/johrstrom/ondemand/misc/config/ondemand.d"
```

In that directory drop a `ondemand.yml` file (or any name you like, but should be .yml extension) with these configs.

```yaml
csp_enabled: true
csp_report_only: true
```

> From what I've read that inline javascript may be an issue as well

Yes, we should be able to follow this pattern - but I always have to re-remember if this works for sure (I'm not 100% sure it does or at least I have to re-check it every time I run across it).

If it has a nonce it should be OK (you have to check the raw HTML to see it, browsers generally remove it when your navigating the DOM i've found)
https://github.com/OSC/ondemand/blob/8ca8e7faa5f76bedd5b297e674f3b627b5778aca/apps/dashboard/app/views/support_ticket/email_service_template.html.erb#L3